### ISSUE TITLE
Push Agent status to server

### ISSUE THREAD

**Issue by MarcosDY** [spiffe/spire#3901]
Add a new package to push status to SPIRE Server and keep the list of tainted keys
The list of tainted keys must be kept here and propagated to Agent Sync and AgentSVID rotator code

Depends on #3898

**Comment by MarcosDY** [comment]
to keep this issue alive

**Comment by MarcosDY** [comment]
This is no longer needed
